### ISSUE TITLE
How to wrap a hook with a Provider with prop passing?

### ISSUE THREAD

**Issue by acrowell1-chwy** [testing-library/react-hooks-testing-library#965]
### What is your question:

I have seen the following which was helpful: https://github.com/testing-library/react-hooks-testing-library/issues/23
This question steams further from that with regards to passing `initialProps`.  
I have updated React and DOM (`"react": "18.2.0", "react-dom": "18.2.0",`) and the React Testing Library (`"@testing-library/react": "14.0.0",`) to the most recent versions. I have followed that pattern, but notice that the props passed into Provider are only available via `children`. Please see the example below (borrowed from the tests within the library https://github.com/testing-library/react-hooks-testing-library/blob/1e01273374af4e48a0feb1f2233bf6c76d742167/src/__tests__/useContext.test.tsx): 

```Javascript 
 test('should update value in context when props are updated', () => {
    const TestContext = createContext('foo');

    const wrapper = ({
      current,
      children,
    }: {
      current: string;
      children: any;
    }) => {
      console.log({ current });
      console.log({ children });
      console.log('nested', children.props.renderCallbackProps);

      return (
        <TestContext.Provider value={current}>{children}</TestContext.Provider>
      );
    };

    const { result, rerender } = renderHook(() => useContext(TestContext), {
      wrapper,
      initialProps: {
        current: 'bar',
      },
    });

    rerender({ current: 'baz' });

    expect(result.current).toBe('baz');
  });

// console.logs

console.log
{ current: undefined }

console.log
{
  children: {
    '$$typeof': Symbol(react.element),
    type: [Function: TestComponent],
    key: null,
    ref: null,
    props: { renderCallbackProps: [Object] },
    _owner: null,
    _store: {}
  }
}

console.log
nested { current: 'baz' }


```

Was this the intended change? Are we supposed to only pass `children` to a `wrapper` and then desconstruct those props  from it? Can we update the tests or docs to represent the recommended pattern?

**Comment by acrowell1-chwy** [comment]
@mpeyper I saw you answer in the previous question. Tagging you here for an extension of it.

**Comment by mpeyper** [comment]
Hey @acrowell1-chwy, just want to double check that you have stopped using this library (`@testing-library/react-hooks' and migrated to `@testing-library/react`), right?

If that’s the case, the wrapper props are handled differently than they were here so our tests and docs do not represent how they intend it to be used.

Unfortunately, documenting the difference of where I got up to in the [“never released” migration guide](https://github.com/testing-library/react-hooks-testing-library/blob/chore/migration-guide/MIGRATION_GUIDE.md#wrapper-props).

**Comment by acrowell1-chwy** [comment]
@mpeyper you are correct, I am using `renderHook` from `@testing-library/react`. Yes, please move this to an issue. Thank you for confirming!
